docs/src/appendix/chisel-plugin.md

```diff
@@ -0,0 +1,80 @@
+# Notes on the Compiler Plug-in
```

Comment:
you need to do an editing pass on this file actually looking at it in a markdown viewer, many issues in `.md` format

Reply:
I made this a bit cleaner, it's not too thorough but has some better debugging hints. Used IntelliJ's md rendering.
I hope that is sufficient.